From 1f7100afd5155470f0db8c2d535217862c886b47 Mon Sep 17 00:00:00 2001
From: Jeffrey Clark <h0tw1r3@users.noreply.github.com>
Date: Sat, 3 Aug 2024 22:50:35 +0000
Subject: [PATCH 3/4] (ospuppet) support ref gem location_for

---
 Gemfile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/Gemfile b/Gemfile
index 06528f2..f90fa35 100644
--- a/Gemfile
+++ b/Gemfile
@@ -2,7 +2,11 @@ source ENV['GEM_SOURCE'] || "https://rubygems.org"
 
 def location_for(place)
   if place =~ /^((?:git[:@]|https:)[^#]*)#(.*)/
-    [{ :git => $1, :branch => $2, :require => false }]
+    if $2[0] == '@'
+      [{ :git => $1, :ref => $2[1..], :require => false }]
+    else
+      [{ :git => $1, :branch => $2, :require => false }]
+    end
   elsif place =~ /^file:\/\/(.*)/
     ['>= 0', { :path => File.expand_path($1), :require => false }]
   else
-- 
2.25.1

